### PR TITLE
Always make the Save & Continue button in the VBA flow active

### PR DESCRIPTION
**Pull request by pecanoro**
### Details
I refactor the view a bit so it displays all messages in case someone clicks the continue button without filling out the information. 

The only part that I am not sure about is that all errors are hidden when one field is modified and there is more than one error, should I change this behavior and remove only the corresponding one?

### Fixed Issues
$ https://github.com/Expensify/App/issues/5182

### `bank-account/new` / Plaid flow 
1. Navigate to `/bank-account/new`
2. Select "Log into your bank"
3. Enter plaid test credentials (Bank: Chase, User: user_good, Pass: pass_good) **Note:** This will only work on native when on the staging secure server which we can switch on in `settings/preferences`.
4. When on this screen verify that you can press "Save & Continue" and the button is not disabled.
5. Press the button
6. Verify that a modal appears with a message
7. Verify that the Choose an account field is highlighted in red

### `bank-account/new` / Add Manual flow
1. Navigate to `/bank-account/new`
2. Select "Connect manually"
3. Verify the "Save & Continue" button is not disabled
4. De-select the "I accept" checkbox so it is not checked
4. Press the "Save & Continue" button
5. Verify the modal appears
6. Verify that the checkbox is outlined in red border
7. Enter plaid test account info (Routing: `011401533`, Account: `1111222233331111`) **Note:** This will only work on native when on the staging secure server which we can switch on in `settings/preferences`.
8. Check the checkbox
9. Verify the red border is cleared
9. Press "Save & Continue"
10. Verify we advance to the next step

### `bank-account/company`

1. Verify Save & Continue button is not disabled
2. Press the button
3. Verify all fields are highlighted in red (**Note:** Not all fields will have messages under them).
4. Enter information into all the fields 
5. Verify the errors are cleared from all fields
6. Press "Save & Continue"

### `bank-account/personal-information`

1. Verify the "Save & Continue" button is not disabled
2. Press the button
3. Verify the modal appears
4. Verify the Checkbox is highlighted and a message displayed under it.
5. Check the checkbox and verify the error clears
6. Sumbit the form again and verify that other missing fields are highlighted
7. Fill out all fields except the "State" and "City" fields
8. Submit the form
11. Verify modal appears + State selector is highlighted + text appears below the field
12. Select a state and verify the error clears
11. Submit again
9. Verify modal + error text appears for the City field
10. Enter a city and verify the error clears 
11. Submit again

### `bank-account/contract`

1. Verify the "Save & Continue" button is not disabled
2. Check "I own more than 25% of..."
2. Press the button
3. Verify modal + terms and conditions checkbox is highlighted + message displayed under
4. Check the box and verify the error clears
5. Submit again
6. Verify the "certify true and accurate" error appears + checkbox is highlighted.
7. Check the box and verify the error clears

### Tested On

- [x] Web
- [x] Mobile Web
- [x] Desktop
- [x] iOS
- [x] Android

### CODE REVIEW

**Comment by marcaaron** [comment]
Still testing on native to make sure things look OK but going to move this out of draft so we can get reviews in.

File: src/components/AddPlaidBankAccount.js [hunk 1]
```diff
@@ -111,7 +113,28 @@ class AddPlaidBankAccount extends React.Component {
...
     }
 
+    /**
+     * @returns {Object}
+     */
+    getErrors() {
+        return lodashGet(this.props, ['reimbursementAccount', 'errors'], {});
+    }
+
+    validate() {
```
Comment (Jag96) [inline src/components/AddPlaidBankAccount.js:123]: Add jsdocs

File: src/components/AddPlaidBankAccount.js [hunk 2]
```diff
@@ -111,7 +113,28 @@ class AddPlaidBankAccount extends React.Component {
         return lodashGet(this.props.plaidBankAccounts, 'accounts', []);
     }
 
+    /**
+     * @returns {Object}
+     */
+    getErrors() {
```
Comment (Jag96) [inline src/components/AddPlaidBankAccount.js:119]: This looks like its repeated in a few places, can we move it to ReimbursementAccountUtils or something similar?

File: src/components/AddPlaidBankAccount.js [hunk 3]
```diff
@@ -163,13 +186,22 @@ class AddPlaidBankAccount extends React.Component {
                                     label={this.props.translate('addPersonalBankAccountPage.chooseAccountLabel')}
                                     onChange={(index) => {
                                         this.setState({selectedIndex: Number(index)});
+                                        const errors = this.getErrors();
+                                        if (!errors.selectedBank) {
+                                            return;
+                                        }
+
+                                        const newErrors = {...errors};
```
Comment (Jag96) [inline src/components/AddPlaidBankAccount.js:194]: Is the destructuring necessary here? Why can't we just delete the `selectedBank` property from `errors` and pass that? Same question for the BankAccountStep.js addition.

File: src/components/Checkbox.js [hunk 4]
```diff
@@ -11,20 +11,29 @@ const propTypes = {
...
 };
 
 const Checkbox = ({
     isChecked,
     onPress,
+    hasError,
 }) => (
     <Pressable onPress={() => onPress(!isChecked)}>
-        <View style={[styles.checkboxContainer, isChecked && styles.checkedContainer]}>
+        <View style={[styles.checkboxContainer, isChecked && styles.checkedContainer, hasError && styles.borderColorDanger]}>
```
Comment (Jag96) [inline src/components/Checkbox.js:29]: Can you move the style prop to its own line and put each style on a new line? There's an example in the `// Good` section of https://github.com/Expensify/App/blob/main/STYLING.md#when-to-create-a-new-style

File: src/components/InlineErrorText.js [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    }
+
+    return (
+        <Text style={[styles.formError, styles.mt1]}>{props.children}</Text>
+    );
+};
+
+InlineErrorText.propTypes = propTypes;
+InlineErrorText.defaultProps = defaultProps;
+export default InlineErrorText;
```
Comment (Jag96) [inline src/components/InlineErrorText.js:28]: Add displayName here: https://github.com/Expensify/App/blob/main/STYLE.md#stateless-component-style

File: src/pages/ReimbursementAccount/BankAccountStep.js [hunk 6]
```diff
@@ -80,21 +80,29 @@ class BankAccountStep extends React.Component {
         return errors[inputKey] ? this.props.translate(this.errorTranslationKeys[inputKey]) : '';
     }
 
+    /**
+     * @param {String} inputKey
+     */
+    clearError(inputKey) {
```
Comment (Jag96) [inline src/pages/ReimbursementAccount/BankAccountStep.js:86]: Looks like this is also duplicated in CompanyStep

**Review by Jag96 — CHANGES_REQUESTED** [review]

File: src/components/AddPlaidBankAccount.js (hunk 2 above)
Comment (marcaaron) [reply src/components/AddPlaidBankAccount.js:119]: Yes! Thanks for this suggestion. I was envisioning this logic would eventually get abstracted into a component somehow. But until we figure that out what that will look like... probably best to have all of these repeated methods in lib.

File: src/components/AddPlaidBankAccount.js (hunk 3 above)
Comment (marcaaron) [reply src/components/AddPlaidBankAccount.js:194]: This SO post explain a bit about why https://stackoverflow.com/questions/37358135/can-i-delete-props-in-react

But the basic idea is that you want your props to be fed from whatever state management you're using (in this case Onyx) and should avoid mutating them (errors is a reference to the errors object in props). So we're making a copy and then deleting the property from there and then setting the errors again.

Comment (Jag96) [reply src/components/AddPlaidBankAccount.js:194]: Ah ok makes sense 👍

File: src/languages/es.js [hunk 7]
```diff
@@ -369,40 +369,42 @@ export default {
...
+            companyType: 'Ingresa un tipo de compañía válido',
+            tooManyAttempts: 'Debido a la gran cantidad de intentos de inicio de sesión, esta opción se ha desactivado temporalmente durante 24 horas. Vuelve a intentarlo más tarde o introduzca los detalles manualmente.',
+            address: 'Ingresa una dirección válida',
+            dob: 'Ingresa una fecha de nacimiento válida',
             age: 'Los solicitantes deben ser mayores de 18 años',
-            ssnLast4: 'Ingrese los últimos 4 dígitos del número de seguro social',
-            noDefaultDepositAccountOrDebitCardAvailable: 'Por favor agregue una cuenta bancaria para depósitos o una tarjeta de débito',
+            ssnLast4: 'Ingresa los últimos 4 dígitos del número de seguro social',
+            firstName: 'Ingresa una primer nombre válida',
+            lastName: 'Ingresa una apellido válida',
```
Comment (marcaaron) [inline src/languages/es.js:406]: Oh yeah also I did not get this checked in espanol ... one sec...

**Comment by marcaaron** [comment]
Updated and re-tested! I liked the idea to abstract those methods so here's how I did it. To make it so we don't have to pass props over and over again or reference `ReimbursementAccountUtils` in the component logic we can bind it to the component like this in the constructor...

https://github.com/Expensify/App/blob/28015cb8d829ec71b529a4d2ac338ec7aebae040/src/components/AddPlaidBankAccount.js#L102-L103

I don't _love_ this because it's kind of "mixin"-y and there's probably a more "react"-y way to do this,  but it'll keep logic centralized while we research a better alternative.

**Review by Jag96 — COMMENTED** [review]
When testing Personal Information, it looks like only the personal address field is highlighted if I submit the form w/ just the checkbox checked

image

Then if I fill out the address field and submit again, only the state field shows as highlighted, then after choosing a state and submitting only the city field shows as highlighted. Should they all show as highlighted?

**Comment by marcaaron** [comment]
>Then if I fill out the address field and submit again, only the state field shows as highlighted, then after choosing a state and submitting only the city field shows as highlighted. Should they all show as highlighted?

We are going to follow up later to handle multiple fields at once, but punted it to prioritize disabling the save & continue button.

**Comment by marcaaron** [comment]
Basically this issue will continue after launch -> https://github.com/Expensify/App/issues/4785

**Review by Jag96 — APPROVED** [review]
LGTM, good stuff!

**Pull request merged**